src/SdkCommon/ClientRuntime/ClientRuntime/ServiceClient.cs

```diff
@@ -251,10 +252,9 @@ private string FrameworkVersion
             "CA2000:Dispose objects before losing scope",
             Justification = "The created objects should be disposed on caller's side")]
         protected ServiceClient()
-            : this(CreateRootHandler())
-        {
-        }
-
+            : this(serviceHttpClient: null, rootHandler: CreateRootHandler(), disposeHttpClient: true, delHandlers: null) { }
```

Comment:
CreateRootHandler is called by this constructor.  This code creates a HttpClientHandler or WebRequestHandler that is then never disposed.  Please see https://github.com/Azure/azure-sdk-for-net/issues/3712